pkg/frontend/tgt/frontend.go

```diff
@@ -70,10 +74,18 @@ func (t *Tgt) Startup(rwu types.ReaderWriterUnmapperAt) error {
 
 	t.isUp = true
 
+	// If the engine failed during a snapshot, we may have left a frozen filesystem. This is a good opportunity to
+	// attempt to unfreeze it.
+	t.attemptUnfreezeFilesystem()
+
 	return nil
 }
 
 func (t *Tgt) Shutdown() error {
+	// If the engine is shutting down during a snapshot (in the preparation phase, before the snapshot operation obtains
+	// a lock) we may have left a frozen filesystem. This is a good opportunity to unfreeze it.
+	t.attemptUnfreezeFilesystem()
```

Comment:
Should we put this logic into [func (d *LonghornDevice) Shutdown()](https://github.com/longhorn/longhorn-engine/blob/a807f0fd6bfd9c4700f2c19808038e87a2ab814e/vendor/github.com/longhorn/go-iscsi-helper/longhorndev/dev.go#L177) instead?

Reply:
See my reply to your similar comment about `startScsiDevice`. I will keep the logic in `longhorn-engine` because of the new location we will execute it from.